src/prefect/tasks.py

```diff
@@ -1027,7 +1033,7 @@ def map(
             # TODO: Make this non-blocking once we can return a list of futures
             # instead of a list of task runs
             return [
-                run_coro_as_sync(self.create_run(parameters=parameters))
+                run_coro_as_sync(self.create_run(parameters=parameters, deferred=True))
```

Comment:
Do we know enough here to say it should definitely be a deferred run?  Is the lack of a `flow_run_context` still enough to say that definitively?

Reply:
Not really. We have an unsolved question about what to do with map(). Map always submits, and `submit()` only works in flows, so really `map()` shouldn't work outside of a flow. We need something like `map_apply_async` that only defers mapped runs.

Reply:
I was wondering the same thing.  Unfortunately it's called `.map` in celery and in `multiprocessing` (although in multiprocessing it's on the `Pool`, not the function)

Reply:
Maybe we could add a `defer` kwarg so that's explicit when a `.map` call will create background tasks? How to handle this is worth noodling on for sure.